crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -836,7 +836,12 @@ impl<'a, 'b> BlankLinesChecker<'a, 'b> {
             // Allow groups of one-liners.
             && !(state.follows.is_any_def() && line.last_token != TokenKind::Colon)
             && !state.follows.follows_def_with_dummy_body()
-            && matches!(state.class_status, Status::Inside(_))
+            // Only apply to functions that are "immediately within" a class (not nested within other functions/blocks)
```

Comment:
Can we add the reasoning why we only trigger for functions directly within a class? 

What about

```py

class Foo:
	if True:
		def test(): pass

```